src/appsync.ts

```diff
@@ -210,63 +250,89 @@ export class AppsyncResolver<
                   )
             );
 
-            if (expr.kind === "ExprStmt" && expr.expr.kind === "CallExpr") {
-              return createStage(expr.expr);
-            } else if (
-              expr.kind === "ReturnStmt" &&
-              expr.expr.kind === "CallExpr"
-            ) {
+            if (stmt.kind === "ExprStmt") {
+              return createStage(getCall(stmt.expr));
+            } else if (stmt.kind === "ReturnStmt") {
               return createStage(
-                expr.expr,
-                appsync.MappingTemplate.fromString(
-                  `#set( $context.stash.return__flag = true )
-#set( $context.stash.return__val = $context.result )
+                getCall(stmt.expr),
+                `#set( $context.stash.return__flag = true )
+#set( $context.stash.return__val = ${getResult(stmt.expr)} )
 {}`
-                )
               );
-            } else if (
-              expr.kind === "VariableStmt" &&
-              expr.expr?.kind === "CallExpr"
-            ) {
-              const responseMappingTemplate =
-                appsync.MappingTemplate.fromString(
-                  `#set( $context.stash.${expr.name} = $context.result )\n{}`
-                );
-
-              return createStage(expr.expr, responseMappingTemplate);
+            } else if (stmt.kind === "VariableStmt" && stmt.expr) {
+              return createStage(
+                getCall(stmt.expr),
+                `#set( $context.stash.${stmt.name} = ${getResult(
+                  stmt.expr
+                )} )\n{}`
+              );
             } else {
               throw new Error(
                 `only a 'VariableDecl', 'Call' or 'Return' expression may call a service`
               );
             }
+            function getCall(expr: Expr): CallExpr {
```

Comment:
doc? Searched up the tree for a call, or ref, or something. Also this function seems really specific to the current integrations of lambda and dynamo tables.

Reply:
Will add docs and explain the cases it is solving. I want to clean this all up but my first priority is to fix bugs and merge step functions before diverging too far.